### PR TITLE
[doc]: update doc react-hook-form

### PR DESCRIPTION
**Pull request by K-Chow**
## Summar
Update document about spike react-hook-form

### CODE REVIEW

File: docs/src/content/docs/en/spikes/tech-spikes-react-hook-form.mdx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+//...
+
+```
+
+### **4. Bind form controls**
+
+There are two ways to bind form controls:
+
+1. Use the register method, which is used for generic components and can simply bind a form component to the form.
```
Comment (mrcuriosity-tw) [inline docs/src/content/docs/en/spikes/tech-spikes-react-hook-form.mdx:137]: here tag `1.`, while line 150 tags `1.` again

File: docs/src/content/docs/en/spikes/tech-spikes-react-hook-form.mdx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+</Container>
+
+```
+
+In such scenarios, how do we easily access the parent component scope within ChildA and ChildB components?
+
+react-hook-form provides the useFormContext syntax for accessing the nearest parent form scope.
+
+```jsx
+export default const ChildA = () => {
```
Comment (mrcuriosity-tw) [inline docs/src/content/docs/en/spikes/tech-spikes-react-hook-form.mdx:238]: wrong syntax?
```
const a = 1;
export default a;
// or
export const a = 1;
```

File: docs/src/content/docs/en/spikes/tech-spikes-react-hook-form.mdx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+	const { setError, getValues } = useFormContext();
+	// do something
+}
+
+```
+
+Through this access method, we can encapsulate some custom form components around react-hook-form, and even combine a series of components into a complex business component, triggering validation and form behavior through manual onChange event.
+
+```jsx
+export default const FormTextField = ({name}) => {
```
Comment (mrcuriosity-tw) [inline docs/src/content/docs/en/spikes/tech-spikes-react-hook-form.mdx:248]: Ditto.

**Review by mrcuriosity-tw — APPROVED** [review]
LGTM.

**Comment by guzhongren** [comment]
https://npmtrends.com/react-final-form-vs-react-hook-form

Add this to show why choose yours

**Pull request merged**